src/main/scala/com/wavesplatform/UtxPool.scala

```diff
@@ -55,6 +89,23 @@ class UtxPool(
     }
   }
 
+  def portfolio(addr: Address): Portfolio = {
```

Comment:
Here you reduce ALL transaction diffs on every request.
Condider `Map[Address,List[TransactionId]]` to be able to fold only nessessary txs.